docs/data-companion/pruning.md

```diff
@@ -0,0 +1,161 @@
+regular gRPC endpoint and require separate configuration and activation. These "privileged" services
+have the ability to manipulate the storage on the node. Therefore, only operators who have privileged access
+to the server should be allowed to use them.
+
+## Pruning configuration
+
+You need to ensure that the pruning for data companion is enabled in the configuration to allow the data companion
+influence the node pruning mechanism.
+
+In order to do that, please ensure that in the `[storage.pruning.data_companion]` section of the CometBFT's configuration
+file, the property `enabled` is set to `true`:
+
+```
+[storage.pruning.data_companion]
+
+# Whether automatic pruning respects values set by the data companion. Disabled
+# by default. All other parameters in this section are ignored when this is
+# disabled.
+#
+# If disabled, only the application retain height will influence block pruning
+# (but not block results pruning). Only enabling this at a later stage will
+# potentially mean that blocks below the application-set retain height at the
+# time will not be available to the data companion.
+enabled = true
+```
+
+In order to avoid unwanted pruning of data when the data companion is activated, it is possible to define the initial
+retain height for block and block results in the configuration file. This configuration ensures that the necessary data
+is retained and not removed in an undesirable way.
+
+For example, you can change the values in this part of the configuration:
+```
+# The initial value for the data companion block retain height if the data
+# companion has not yet explicitly set one. If the data companion has already
+# set a block retain height, this is ignored.
+initial_block_retain_height = 0
+
+# The initial value for the data companion block results retain height if the
+# data companion has not yet explicitly set one. If the data companion has
+# already set a block results retain height, this is ignored.
+initial_block_results_retain_height = 10
+```
+
+## Retain Height parameter
+
+The pruning API uses two crucial parameters that allow for efficient data storage management. The "block retain height"
+pruning parameter specifies the height to which the node will preserve blocks. It is important to note that this
+parameter differs from the application block retain height, which the application sets in response to ABCI commit messages.
+
+The node will preserve blocks up to the lowest value between the data companion retain height and the application block retain height.
+
+The "block results retain height" pruning parameter is the second crucial aspect of the pruning API.
+This parameter determines the height up to which the node will keep block results. By retaining block results to a
+certain height, the node can efficiently manage its storage and optimize its performance.
+
+With these two parameters, the pruning service ensures that the node efficiently manages its data storage.
+
+### Block Retain Height
+
```

Comment:
I would note here that this retain height can also be influenced by the application. AS in- it is the only retain height of which the application has control. The application controls this via appRetainHeight into which the operator can get insight via a gRPC call. Note that , at any point in time, only the lowest retain height is accepted. By default the application block retain height and the data companion retain height are set to 0. This is done in order to prevent any one of them pruning the data while the other has not indicated it is ok to do so.

Reply:
I had added some information related to this but I'll augment that with info from this comment, thanks.